cmd/kube-apiserver/app/server.go

```diff
@@ -551,37 +552,39 @@ func (s *APIServer) Run(_ []string) error {
 	}
 
 	config := &master.Config{
-		StorageDestinations:       storageDestinations,
-		StorageVersions:           storageVersions,
-		EventTTL:                  s.EventTTL,
-		KubeletClient:             kubeletClient,
-		ServiceClusterIPRange:     &n,
-		EnableCoreControllers:     true,
-		EnableLogsSupport:         s.EnableLogsSupport,
-		EnableUISupport:           true,
-		EnableSwaggerSupport:      true,
-		EnableProfiling:           s.EnableProfiling,
-		EnableWatchCache:          s.EnableWatchCache,
-		EnableIndex:               true,
-		APIPrefix:                 s.APIPrefix,
-		APIGroupPrefix:            s.APIGroupPrefix,
-		CorsAllowedOriginList:     s.CorsAllowedOriginList,
-		ReadWritePort:             s.SecurePort,
-		PublicAddress:             s.AdvertiseAddress,
-		Authenticator:             authenticator,
-		SupportsBasicAuth:         len(s.BasicAuthFile) > 0,
-		Authorizer:                authorizer,
-		AdmissionControl:          admissionController,
-		APIGroupVersionOverrides:  apiGroupVersionOverrides,
-		MasterServiceNamespace:    s.MasterServiceNamespace,
-		ClusterName:               s.ClusterName,
-		ExternalHost:              s.ExternalHost,
-		MinRequestTimeout:         s.MinRequestTimeout,
-		ProxyDialer:               proxyDialerFn,
-		ProxyTLSClientConfig:      proxyTLSClientConfig,
-		Tunneler:                  tunneler,
-		ServiceNodePortRange:      s.ServiceNodePortRange,
-		KubernetesServiceNodePort: s.KubernetesServiceNodePort,
+		Config: &genericapiserver.Config{
+			StorageDestinations:       storageDestinations,
+			StorageVersions:           storageVersions,
+			EventTTL:                  s.EventTTL,
+			KubeletClient:             kubeletClient,
+			ServiceClusterIPRange:     &n,
+			EnableLogsSupport:         s.EnableLogsSupport,
+			EnableUISupport:           true,
+			EnableSwaggerSupport:      true,
+			EnableProfiling:           s.EnableProfiling,
+			EnableWatchCache:          s.EnableWatchCache,
+			EnableIndex:               true,
+			APIPrefix:                 s.APIPrefix,
+			APIGroupPrefix:            s.APIGroupPrefix,
```

Comment:
If its customizable in core master, then it needs to be customizable for all api servers.
If its not required, I can remove it everywhere.